common/testing/testhooks/noop_impl.go

```diff
@@ -3,51 +3,38 @@
 package testhooks
 
 import (
-	"context"
-
 	"go.uber.org/fx"
 )
 
 var Module = fx.Options(
-	fx.Provide(func() (_ TestHooks) { return }),
+	fx.Provide(NewTestHooks),
```

Comment:
what was wrong with the interface/concrete switch? an interface takes 16 bytes, a concrete struct{} takes 0. yeah, it may not be noticeable but why change it?

Reply:
I was playing around with some options; the size wasn't top of mind for me here. I'll revert that, no problem.